core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -929,6 +982,56 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
         }
     }
 
+    fun signDefensivePact(duration: Int) {
+        //Note: These modifiers are additive to the freindship modifiers
+        setModifier(DiplomaticModifiers.DefensivePact, 30f)
+        otherCivDiplomacy().setModifier(DiplomaticModifiers.DefensivePact, 30f)
+        diplomaticStatus = DiplomaticStatus.DefensivePact;
+        otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.DefensivePact;
+
+        for (thirdCiv in getCommonKnownCivs().filter { it.isMajorCiv() }) {
+            thirdCiv.addNotification("[${civInfo.civName}] and [$otherCivName] have signed the Defensive Pact!",
+                NotificationCategory.Diplomacy, civInfo.civName, NotificationIcon.Diplomacy, otherCivName)
+            thirdCiv.getDiplomacyManager(civInfo).setDefensivePactBasedModifier()
+        }
+        civInfo.isAtWar()
+        //Have them join our wars
+        for (newWar in civInfo.diplomacy.values.filter { it.diplomaticStatus == DiplomaticStatus.DefensiveWar && !it.otherCiv().isDefeated()
+            && !otherCivDiplomacy().civInfo.isAtWarWith(it.otherCiv())
+        }) {
+                otherCiv().diplomacy.values.find { it.otherCiv() == newWar.otherCiv() }?.declareWar(true)
+        }
+        //Join thier wars
+        for (newWar in otherCiv().diplomacy.values.filter { it.diplomaticStatus == DiplomaticStatus.DefensiveWar && !it.otherCiv().isDefeated()
+            && !civInfo.isAtWarWith(it.otherCiv())}) {
+                civInfo.diplomacy.values.find { it.otherCiv() == newWar.otherCiv() }?.declareWar()
+        }
+
+        // Ignore contitionals as triggerCivwideUnique will check again, and that would break
+        // UniqueType.ConditionalChance - 25% declared chance would work as 6% actual chance
+        for (unique in civInfo.getTriggeredUniques(UniqueType.TriggerUponSigningDefensivePact, StateForConditionals.IgnoreConditionals))
+            UniqueTriggerActivation.triggerCivwideUnique(unique, civInfo)
+        for (unique in otherCiv().getTriggeredUniques(UniqueType.TriggerUponSigningDefensivePact, StateForConditionals.IgnoreConditionals))
+            UniqueTriggerActivation.triggerCivwideUnique(unique, otherCiv())
+    }
+
+    private fun setDefensivePactBasedModifier() {
+        removeModifier(DiplomaticModifiers.SignedDefensivePactWithOurAllies)
+        removeModifier(DiplomaticModifiers.SignedDefensivePactWithOurEnemies)
+        for (thirdCiv in getCommonKnownCivs()
+            .filter { it.getDiplomacyManager(civInfo).hasFlag(DiplomacyFlags.DefensivePact) }) {
+            //Note: These modifiers are additive to the freindship modifiers
+            when (otherCiv().getDiplomacyManager(thirdCiv).relationshipIgnoreAfraid()) {
```

Comment:
When should be for the amount, since the action is the same for all skins except 0
Would result in cleaner code

Reply:
setFriendshipBasedModifier() uses this way of deciding the modifier. The problem is that there is a difference between DiplomaticModifiers.SignedDefensivePactWithOurEnemies and DiplomaticModifiers.SignedDefensivePactWithOurAllies. The revised code would require two when statements unless there is another way of doing it. 

`
val relationshipLevel =  otherCiv().getDiplomacyManager(thirdCiv).relationshipIgnoreAfraid()
            addModifier(when (relationshipLevel) {
                RelationshipLevel.Unforgivable, RelationshipLevel.Enemy -> DiplomaticModifiers.SignedDefensivePactWithOurEnemies
                RelationshipLevel.Friend, RelationshipLevel.Ally -> DiplomaticModifiers.SignedDefensivePactWithOurAllies
                else -> DiplomaticModifiers.SignedDefensivePactWithOurAllies }, when (relationshipLevel) {
                RelationshipLevel.Unforgivable -> -30f
                RelationshipLevel.Enemy -> -15f
                RelationshipLevel.Friend ->  5f
                RelationshipLevel.Ally -> 15f
                else -> {0f}
            })
`

If you want me to implement this change, I would probably change setFriendshipBasedModifier() as well.

Reply:
I agree, having the 'when' there set a variable would my much cleaner